storage/src/test/java/tech/pegasys/teku/storage/server/AbstractDatabaseTest.java

```diff
@@ -490,6 +496,53 @@ public void testShouldRecordFinalizedBlocksAndStatesInBatchUpdate()
     testShouldRecordFinalizedBlocksAndStates(StateStorageMode.ARCHIVE, true);
   }
 
+  @Test
+  public void shouldRecreateMerkleTreeFromDatabase() {
+    final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+    DepositWithIndex firstDeposit = dataStructureUtil.randomDepositWithIndex(1L);
+    DepositWithIndex secondDeposit = dataStructureUtil.randomDepositWithIndex(2L);
+    eth1DepositChannel.addEth1Deposit(firstDeposit);
+    eth1DepositChannel.addEth1Deposit(secondDeposit);
+
+    assertThat(eth1DepositChannel.getDeposits().get(0)).isEqualTo(firstDeposit);
+    assertThat(eth1DepositChannel.getDeposits().get(1)).isEqualTo(secondDeposit);
+    MerkleTree m = database.getMerkleTree();
+    assertThat(m.getProof(firstDeposit.getData().hash_tree_root())).isNotEmpty();
+    assertThat(m.getProof(secondDeposit.getData().hash_tree_root())).isNotEmpty();
```

Comment:
We should check the root hash of the tree matches. 
```suggestion
     MerkleTree expectedTree = new OptimizedMerkleTree(Constants.DEPOSIT_CONTRACT_TREE_DEPTH);
    expectedTree.add(firstDeposit.getData().hash_tree_root());
    expectedTree.add(secondDeposit.getData().hash_tree_root());
    assertThat(database.getMerkleTree().getRoot()).isEqualTo(expectedTree.getRoot());
```

The nice thing about Merkle trees is if the root matches, everything is exactly the same - no missing items and no extra items either.